Support isolation.level config for Kafka consumer

lhooss opened the pull request:
### Description
This pull request adds support for the `isolation.level` Kafka consumer configuration in Data Prepper.

The following changes were made:

- Introduced a new method `getIsolationLevel()` in the `TopicConsumerConfig` interface.

- Implemented the new property `isolation_level` with a default value of `"read_uncommitted"` in both:

    - `SourceTopicConfig` (Kafka source plugin)

    - `BufferTopicConfig` (Kafka buffer plugin)

- Ensured backward compatibility by setting sensible defaults and integrating with existing configuration parsing logic.

- This allows users to configure Data Prepper to read only committed messages by setting `isolation_level: read_committed` in their pipeline YAML file.

This is especially useful when working with transactional Kafka producers and ensures message consistency in distributed systems.

### Issues Resolved
Resolves #5896
 
### Check List
- [x] New functionality includes testing.
- [ ] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

dlvenable commented on data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/buffer/BufferTopicConfig.java:
```diff
@@ -117,6 +117,9 @@ class BufferTopicConfig extends CommonTopicConfig implements TopicProducerConfig
     @JsonProperty("fetch_min_bytes")
     private ByteCount fetchMinBytes = DEFAULT_FETCH_MIN_BYTES;
 
+    @JsonProperty("isolation_level")
```
Actually, the buffer would need to support transactions. Let's just remove it from `BufferTopicConfig` and make this a source property only.

I understand from #5896 that you are using Kafka as a source.

lhooss commented:
> @lhooss , Thank you for this contribution! I left a couple of comments, but we should be able to get this in once those are done. Thanks!

@dlvenable Thank you for your feedback and for reviewing the PR!
I’ve addressed your comments, please let me know if anything else is needed.

dlvenable commented on data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/IsolationLevel.java:
```diff
@@ -0,0 +1,33 @@
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+public enum IsolationLevel {
```
@lhooss , Thank you for updating to an enum.

Please also add some unit tests for this enum similar to these:

https://github.com/opensearch-project/data-prepper/blob/af5dce7cd019a577aa6ae64a8081cd9beb787dc2/data-prepper-plugins/aggregate-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/aggregate/actions/RateLimiterModeTest.java#L26
